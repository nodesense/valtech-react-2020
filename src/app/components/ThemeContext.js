import React from 'react';

// success is a css class defined in index.css, has green bg
const ThemeContext = React.createContext("success"); 
ThemeContext.displayName = "ThemeContext";
export default ThemeContext;