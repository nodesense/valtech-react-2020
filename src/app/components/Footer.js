import React from 'react';
// for prop validation
import PropTypes from 'prop-types';

import ThemeContext from './ThemeContext';

// functional component
// destructure in the function declaration
// title = props.title
// countries = props.countries
// statements cannot be written inside JSX
const Footer = ( {title, countries, company, children} ) => (
    <div>
        <hr />
        <p>Copyrights @2020, {title}, {company} </p>

        <div>
            {
                countries.map ( (name, i) => (
                    <span key={name} >{i}.{name} </span>
                ))
            }
        </div>

        {children}

        <ThemeContext.Consumer>
            {
                (value) => {
                    console.log('theme value in func', value);
                    return (
                        <button className={value}>
                            Contact Us
                        </button>
                    )
                }
            }
        </ThemeContext.Consumer>

    </div>
)

// keyword
Footer.propTypes = {
    title: PropTypes.string.isRequired,
    countries: PropTypes.array.isRequired,
    company: PropTypes.string // optional
}

//keyword
Footer.defaultProps = {
    company: 'Valtech'
}

 
export default Footer;