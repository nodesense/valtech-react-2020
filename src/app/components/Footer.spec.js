import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';

import ThemeContext from './ThemeContext';

describe ("test for footer", () => {
    it("footer test", () => {
        const component = renderer
            .create(<React.Fragment >
                        <ThemeContext.Provider value='green'>
                            <Footer title="Product app"
                                    countries= { ['IN', 'USA'] }>

                            </Footer>
                        </ThemeContext.Provider>
                    </React.Fragment> 
                )

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})