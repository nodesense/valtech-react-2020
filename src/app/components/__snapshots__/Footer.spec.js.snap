// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for footer footer test 1`] = `
<div>
  <hr />
  <p>
    Copyrights @2020, 
    Product app
    , 
    Valtech
     
  </p>
  <div>
    <span>
      0
      .
      IN
       
    </span>
    <span>
      1
      .
      USA
       
    </span>
  </div>
  <button
    className="green"
  >
    Contact Us
  </button>
</div>
`;
